packages/core/src/behavior.ts

```diff
@@ -0,0 +1,59 @@
+import { ActorRef, Behavior, EventObject, Observer } from '.';
+import { toActorRef } from './Actor';
+import { toEventObject, toObserver } from './utils';
+
+/**
+ * Returns an actor behavior from a reducer and its initial state.
+ *
+ * @param transition The pure reducer that returns the next state given the current state and event.
+ * @param initialState The initial state of the reducer.
+ * @returns An actor behavior
+ */
+export function fromReducer<TState, TEvent extends EventObject>(
+  transition: (state: TState, event: TEvent) => TState,
+  initialState: TState
+): Behavior<TEvent, TState> {
+  return {
+    transition,
+    initialState
+  };
+}
+
+interface SpawnBehaviorOptions {
+  id?: string;
+  parent?: ActorRef<any>;
+}
+
+export function spawnBehavior<TActorEvent extends EventObject, TEmitted>(
+  behavior: Behavior<TActorEvent, TEmitted>,
+  options: SpawnBehaviorOptions = {}
+): ActorRef<TActorEvent, TEmitted> {
+  let state = behavior.initialState;
+  const observers = new Set<Observer<TEmitted>>();
+
+  const actor = toActorRef({
+    id: options.id,
+    send: (event: TActorEvent) => {
```

Comment:
this currently lacks the concept of mailbox/scheduling. It doesn't matter how we achieve this right now but the point is that with this implementation it's possible to start processing a new event in the middle of processing a previous event